hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -383,6 +391,12 @@ public void initTableMetadata() {
     }
 
     if (!exists) {
+      if (metadataWriteConfig.isMetadataAsyncIndex()) {
+        // with async metadata indexing enabled, there can be inflight writers
+        // TODO: schedule indexing only for enabled partition types
```

Comment:
There will be more merge conflicts with https://github.com/apache/hudi/pull/4746. Better to rebase sooner than later. Also, that PR takes care of init for all enabled partitions